packages/flutter/lib/src/material/user_accounts_drawer_header.dart

```diff
@@ -60,75 +62,121 @@ class _AccountPictures extends StatelessWidget {
   }
 }
 
-class _AccountDetails extends StatelessWidget {
+class _AccountDetails extends StatefulWidget {
   const _AccountDetails({
-    Key key,
     @required this.accountName,
     @required this.accountEmail,
     this.onTap,
     this.isOpen,
-  }) : super(key: key);
+  });
 
   final Widget accountName;
   final Widget accountEmail;
   final VoidCallback onTap;
   final bool isOpen;
 
+  @override
+  _AccountDetailsState createState() => new _AccountDetailsState();
+}
+
+class _AccountDetailsState extends State<_AccountDetails> with SingleTickerProviderStateMixin {
+  Animation<double> _animation;
+  AnimationController _controller;
+
+  @override
+  void initState () {
+    super.initState();
+
+    _controller = new AnimationController(
+      value: widget.isOpen ? 1.0 : 0.0,
+      duration: const Duration(milliseconds: 200),
+      vsync: this,
+    );
+    _animation = new CurvedAnimation(
+        parent: _controller,
+        curve: Curves.fastOutSlowIn,
+        reverseCurve: Curves.fastOutSlowIn.flipped,
+      )
+      ..addListener(() => setState(() {
+        // [animation]'s value has changed here.
+      }));
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget (_AccountDetails oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (widget.isOpen) {
+      _controller.forward();
```

Comment:
This will cause the animation to repeatedly restart each time the widget is rebuilt, if the direction didn't change. You probably want to check if the animation is already running, and only start it if it's not running in the right direction. So e.g. in this case you'd want to call forward() if the animation status was dismissed or reverse, and where you call reverse you only want to call it if the status is completed or forward.